src/executor/hash_join_executor.cpp

```diff
@@ -109,19 +120,19 @@ bool HashJoinExecutor::DExecute() {
     // Go over the left tile
     for (auto left_tile_itr : *left_tile) {
       const ContainerTuple<executor::LogicalTile> left_tuple(
-          left_tile, left_tile_itr, &hashed_col_ids);
+          left_tile, left_tile_itr, &left_hashed_col_ids);
 
       // Find matching tuples in the hash table built on top of the right table
       auto right_tuples = hash_table.find(left_tuple);
 
       if (right_tuples != hash_table.end()) {
     	// Not yet supported due to assertion in gettomg right_tuples->first
```

Comment:
Im not sure I understand this. Could you briefly explain what this means.

Reply:
How is `hashed_col_ids` different from `left_hashed_col_ids`? If it is not being used, can we get rid of the variable in hash executor and associated function `hash_executor_->GetHashKeyIds()`?

Reply:
Previously, the hash join executor used the hashed_col_ids from the hash executor, which are column ids of the hash keys in the build side (right_tile), and can be different from the hash key column ids in the probe side (left tile). This fix constructs the column ids correctly from the left table and yes, we can remove `hash_executor_->GetHashKeyIds()`. I will clean it up.